### PR TITLE
Replace System.lineSeparator() with "\n" in test to fix Windows fail

### PR DESCRIPTION
**Pull request by trancexpress**
Fixes: #432
Signed-off-by: Simeon Andreev <simeon.danailov.andreev@gmail.com>

## Author checklist

- [ ] I have thoroughly tested my changes
- [ ] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [ ] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by iloveeclipse** [comment]
This patch below fixes the test on Windows, BUT only if I disable last snippet.
It is simply not found on Windows, IMethod is there but search can't find it.

Here is the diff that is green on Windows:
```
diff --git a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchTests.java b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchTests.java
index 54733fd..4542e30 100644
--- a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchTests.java
+++ b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchTests.java
@@ -4653,3 +4653,3 @@
 			snippet3,
-			snippet4,
+//			snippet4,
 	};
@@ -4659,3 +4659,3 @@
 			"void run() {key=Lp/TestGh432$138;.run()V} [in <anonymous #1> [in run() [in <anonymous #1> [in main(String[]) [in TestGh432 [in TestGh432.java [in p [in src [in gh432MethodReferencesSearchBug]]]]]]]]]",
-			"run() (not open) {key=Lp/TestGh432$162;.run()V} [in <anonymous #1> [in run() [in <lambda #1> [in main(String[]) [in TestGh432 [in TestGh432.java [in p [in src [in gh432MethodReferencesSearchBug]]]]]]]]]",
+//			"run() (not open) {key=Lp/TestGh432$162;.run()V} [in <anonymous #1> [in run() [in <lambda #1> [in main(String[]) [in TestGh432 [in TestGh432.java [in p [in src [in gh432MethodReferencesSearchBug]]]]]]]]]",
 
@@ -4679,4 +4679,4 @@
 			IMethod testMethod = findMethod(p, "p.TestGh432", "missingReference");
-			String foundReferences = searchForMethodReferences(testMethod);
-			assertFalse("Expected search to find references of method: " + testMethod + ", in snippet:\n" + fileContent, foundReferences.isEmpty());
+			String foundReferences = searchForMethodReferences(testMethod).strip();
+			assertFalse(i + " Expected search to find references of method: " + testMethod + ", in snippet:\n" + fileContent, foundReferences.isEmpty());
 			List<String> results = Arrays.asList(foundReferences.split("\n"));
```

Here is the test fail without uncommenting last snippet:

```
junit.framework.AssertionFailedError: 3 Expected search to find references of method: static boolean missingReference(Object) [in TestGh432 [in TestGh432.java [in p [in src [in gh432MethodReferencesSearchBug]]]]], in snippet:
package p;
public class TestGh432 {
  private static void foo(Runnable r) {
    r.run();
  }
  public static void main(String[] args) {
    foo(() -> {
      new Runnable() {
        public void run() {
          java.util.stream.Stream.empty().filter(TestGh432::missingReference).count();
        }
      };
    });
  }
  public static boolean missingReference(Object arg) {
    return true;
  }
}
	at junit.framework.Assert.fail(Assert.java:57)
	at junit.framework.Assert.assertTrue(Assert.java:22)
	at junit.framework.Assert.assertFalse(Assert.java:39)
	at junit.framework.TestCase.assertFalse(TestCase.java:208)
	at org.eclipse.jdt.core.tests.model.JavaSearchTests.testAnonymousTypeMethodReferenceSearchGh432(JavaSearchTests.java:4681)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at junit.framework.TestCase.runTest(TestCase.java:177)
	at org.eclipse.jdt.core.tests.junit.extension.TestCase.runTest(TestCase.java:983)
	at junit.framework.TestCase.runBare(TestCase.java:142)
	at junit.framework.TestResult$1.protect(TestResult.java:122)
	at junit.framework.TestResult.runProtected(TestResult.java:142)
	at junit.framework.TestResult.run(TestResult.java:125)
	at junit.framework.TestCase.run(TestCase.java:130)
	at junit.extensions.TestDecorator.basicRun(TestDecorator.java:24)
	at junit.extensions.TestSetup$1.protect(TestSetup.java:23)
	at junit.framework.TestResult.runProtected(TestResult.java:142)
	at junit.extensions.TestSetup.run(TestSetup.java:27)
	at org.eclipse.jdt.internal.junit.runner.junit3.JUnit3TestReference.run(JUnit3TestReference.java:128)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:529)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:756)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:452)
	at org.eclipse.pde.internal.junit.runtime.RemotePluginTestRunner.main(RemotePluginTestRunner.java:74)
	at org.eclipse.pde.internal.junit.runtime.CoreTestApplication.start(CoreTestApplication.java:28)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:136)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:402)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:596)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1467)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1440)
```

**Comment by trancexpress** [comment]
> This patch below fixes the test on Windows, BUT only if I disable last snippet. It is simply not found on Windows, IMethod is there but search can't find it.

I'm not sure the last snippet passes in general, at least not on current `master`?

File: org.eclipse.jdt.core.tests.model/workspace/gh432MethodReferencesSearchBug/.classpath [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-11"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.MODULE_PATH"/>
```
Comment (iloveeclipse) [inline org.eclipse.jdt.core.tests.model/workspace/gh432MethodReferencesSearchBug/.classpath:4]: What is that and why is this needed?

Comment (trancexpress) [reply org.eclipse.jdt.core.tests.model/workspace/gh432MethodReferencesSearchBug/.classpath:4]: > What is that and why is this needed?

https://github.com/eclipse-jdt/eclipse.jdt.core/pull/433#discussion_r985627128

Comment (iloveeclipse) [reply org.eclipse.jdt.core.tests.model/workspace/gh432MethodReferencesSearchBug/.classpath:4]: Doesn't answer first part of the question. Also I wonder how other tests work that have not checked in .classpath files.

Comment (trancexpress) [reply org.eclipse.jdt.core.tests.model/workspace/gh432MethodReferencesSearchBug/.classpath:4]: > Doesn't answer first part of the question. Also I wonder how other tests work that have not checked in .classpath files.

Did you read the comment? **That** snippet doesn't work without a JRE on the classpath, which is normally not added during tests. The rest of the tests don't run into the problem and so they work.

Comment (iloveeclipse) [reply org.eclipse.jdt.core.tests.model/workspace/gh432MethodReferencesSearchBug/.classpath:4]: OK, you didn't understand my question. 
1) What is **this**: <classpathentry kind="con" path="org.eclipse.jdt.MODULE_PATH"/> nad why it is needed
2) what is special with the concrete test snippet, three other snippets work fine?

Comment (trancexpress) [reply org.eclipse.jdt.core.tests.model/workspace/gh432MethodReferencesSearchBug/.classpath:4]: No idea about `org.eclipse.jdt.MODULE_PATH`, I believe I either copied the `.classpath` of the generated test project and added a JRE to it, or I copied the `.classpath` from another test and added the JRE to it. I can't remember.

> what is special with the concrete test snippet

It has a lambda, and in the lambda body there is an expression that cannot be resolved. The full analysis is listed here: https://github.com/eclipse-jdt/eclipse.jdt.core/pull/433#discussion_r985540992

Comment (iloveeclipse) [reply org.eclipse.jdt.core.tests.model/workspace/gh432MethodReferencesSearchBug/.classpath:4]: Note: the patch still doesn't work on Windows, since you've ignored all my changes :-(
And `org.eclipse.jdt.MODULE_PATH` isn't needed, but I assume the `.classpath` is also a hack and can be replaced by a proper JDT core test framework usage.

Comment (trancexpress) [reply org.eclipse.jdt.core.tests.model/workspace/gh432MethodReferencesSearchBug/.classpath:4]: > Note: the patch still doesn't work on Windows, since you've ignored all my changes :-( And `org.eclipse.jdt.MODULE_PATH` isn't needed, but I assume the `.classpath` is also a hack and can be replaced by a proper JDT core test framework usage.

Adding a JRE to the classpath will be a no small amount of extra code. As already said in the linked comment, adding a `.classpath` is the easiest option. What is the problem with it?

What change did I ignore? The only change is to this line:

`IJavaProject p = setupModuleProject(testProjectName, new String[] {"src"}, new String[0], null);`

Since for whatever reason this line doesn't import the project.

Comment (iloveeclipse) [reply org.eclipse.jdt.core.tests.model/workspace/gh432MethodReferencesSearchBug/.classpath]: See my last push. That works on Windows & Java 11

Comment (trancexpress) [reply org.eclipse.jdt.core.tests.model/workspace/gh432MethodReferencesSearchBug/.classpath]: > See my last push. That works on Windows & Java 11

Works also on Linux, with Java 11 and Java 17.
